pkg/resourceinterpreter/configurableinterpreter/luavm/lua.go

```diff
@@ -53,6 +53,8 @@ func (vm *VM) NewLuaState() (*lua.LState, error) {
 	l.PreloadModule(lua.OsLibName, lifted.SafeOsLoader)
 	// preload kube library. Allows the 'local kube = require("kube")' to work
 	l.PreloadModule(KubeLibName, KubeLoader)
+	// preload string library. Allows the 'local string = require("string")' to work
+	l.PreloadModule(lua.StringLibName, lua.OpenString)
```

Comment:
Can we consider reusing this:

https://github.com/karmada-io/karmada/blob/e4f5b627ef6080daf05cf4c0fc984427b4f9a8d8/vendor/github.com/yuin/gopher-lua/state.go#L1372-L1373

In fact, I wonder why we dropped the setting:

https://github.com/karmada-io/karmada/blob/e4f5b627ef6080daf05cf4c0fc984427b4f9a8d8/pkg/resourceinterpreter/configurableinterpreter/configurable.go#L32

Reply:
Because the lua libraries `debug`, `channel`, `coroutine` are not useful to us.

Reply:
Thanks @ikaven1024 
Our current changes should be reasonable, right?

Reply:
I have confirmed with @chaunceyjiang that the methods provided by the `string` and `math` packages do not have insecure privacy. In addition, according to the comments in argo-cd, insecure methods are in the `os` package and are not involved in this modification.